src/pages/organize/[orgId]/campaigns/[campId]/activities/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { Box } from '@mui/material';
+import { GetServerSideProps } from 'next';
+
+import ActivityList from 'features/campaigns/components/ActivityList';
+import CampaignActivitiesModel from 'features/campaigns/models/CampaignAcitivitiesModel';
+import messageIds from 'features/campaigns/l10n/messageIds';
+import NoActivities from 'features/campaigns/components/NoActivities';
+import { PageWithLayout } from 'utils/types';
+import { scaffold } from 'utils/next';
+import SingleCampaignLayout from 'features/campaigns/layout/SingleCampaignLayout';
+import { useMessages } from 'core/i18n';
+import useModel from 'core/useModel';
+import useServerSide from 'core/useServerSide';
+
+export const getServerSideProps: GetServerSideProps = scaffold(
+  async (ctx) => {
+    const { campId, orgId } = ctx.params!;
+
+    return {
+      props: {
+        campId,
+        orgId,
+      },
+    };
+  },
+  {
+    authLevelRequired: 2,
+    localeScope: ['layout.organize.surveys', 'pages.organizeSurvey'],
+  }
+);
+
+interface CampaignActivitiesPageProps {
+  campId: string;
+  orgId: string;
+}
+
+const CampaignActivitiesPage: PageWithLayout<CampaignActivitiesPageProps> = ({
+  campId,
+  orgId,
+}) => {
+  const messages = useMessages(messageIds);
+  const onServer = useServerSide();
+  const model = useModel(
+    (env) => new CampaignActivitiesModel(env, parseInt(orgId))
+  );
+  const activities = model
```

Comment:
Maybe this filtering could be moved _into_ the model? I'll let you decide!

Reply:
so make two different methods for the different types of filtering (here for campaign id, and standalones on the pther page)?

Reply:
Yes, that's what I was envisioning, but I haven't thought it through so I don't know how much of an improvement that would be. You decide! 😊

Reply:
Yes, that's what I was envisioning, but I haven't thought it through so I don't know how much of an improvement that would be. You decide! 😊

Reply:
will test and see!